src/reporter/github.js

```diff
@@ -114,25 +162,30 @@ export default class GitHub {
 
   async closeIssueWithCommentIfExists({ title, comment }) {
     try {
-      const openedIssue = await this.getIssue({ title, state: GitHub.ISSUE_STATE_OPEN });
+      await this.ensureCacheLoaded();
 
-      if (!openedIssue) {
+      const issue = this.getIssue(title);
+
+      if (!issue || issue.state == GitHub.ISSUE_STATE_CLOSED) {
         return;
       }
 
-      await this.addCommentToIssue({ issue: openedIssue, comment });
-      logger.info(`Added comment to issue #${openedIssue.number}: ${openedIssue.html_url}`);
+      await this.addCommentToIssue({ issue, comment });
+
+      const updatedIssue = await this.updateIssue(issue, { state: GitHub.ISSUE_STATE_CLOSED });
 
-      await this.updateIssue(openedIssue, { state: GitHub.ISSUE_STATE_CLOSED });
-      logger.info(`Closed issue #${openedIssue.number}: ${openedIssue.html_url}`);
+      this.issuesCache.set(updatedIssue.title, updatedIssue);
```

Comment:
Shouldn't the cache update be done in `updateIssue`? I find it very surprising that we have to handle that in the caller.

Reply:
You're absolutely right!

Reply:
And it was already done